daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs

```diff
@@ -168,6 +170,21 @@ compileFile service fp = do
             throwE diag
         Just v -> return v
 
+-- | Parse the supplied file to a ghc ParsedModule.
+parseFile
+    :: IdeState
+    -> NormalizedFilePath
+    -> ExceptT [FileDiagnostic] IO ParsedModule
+parseFile service fp = do
```

Comment:
Why go via the Ide here? Seems almost overkill for parsing a module

Reply:
Seemed like the most straight forward option. Can I call `parseFile` directly without repeating a lot of work to setup state for GHC?

Reply:
Perhaps. Perhaps not... I'm not that sure, but if it seems hard, stick with what you have.